cli_tools/common/utils/storage/resource_location_retriever.go

```diff
@@ -0,0 +1,150 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/domain"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	daisycompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"google.golang.org/api/compute/v1"
+)
+
+var (
+	// for each GCS multi region, regions are sorted by GCE cost in ascending order as of the time of writing
+	// this code.
+	// GCE cost per region: https://cloud.google.com/compute/vm-instance-pricing
+	// GCS multi-regions: https://cloud.google.com/storage/docs/locations#location-mr
+	multiRegions = map[string][]string{
+		"US":   {"us-central1", "us-east1", "us-east4", "us-west1", "us-west2"},
+		"EU":   {"europe-north1", "europe-west1", "europe-west2", "europe-west3", "europe-west4", "europe-west6"},
+		"ASIA": {"asia-east1", "asia-east2", "asia-south1", "asia-southeast1", "asia-northeast1", "asia-northeast2"},
+
+		"EUR4": {"europe-north1", "europe-west4"},
+		"NAM4": {"us-central1", "us-east1"},
+	}
+)
+
+// ResourceLocationRetriever is responsible for retrieving GCE zone to run import in
+type ResourceLocationRetriever struct {
+	Mgce              domain.MetadataGCEInterface
+	ComputeGCEService daisycompute.Client
+}
+
+// NewResourceLocationRetriever creates a ResourceLocationRetriever
+func NewResourceLocationRetriever(aMgce domain.MetadataGCEInterface, cs daisycompute.Client) *ResourceLocationRetriever {
+	return &ResourceLocationRetriever{Mgce: aMgce, ComputeGCEService: cs}
+}
+
+// GetZone retrieves GCE zone to run import in based on imported source file location and available
+// zones in the project.
+// If storageRegion is provided and valid, first zone within that region will be used.
+// If no storageRegion is provided, GCE Zone from the running process
+// will be used.
+func (rlr *ResourceLocationRetriever) GetZone(storageLocation string, project string) (string, error) {
+	zone := ""
+	var err error
+	if storageLocation != "" {
+		// pick a zone from the region where data is stored
```

Comment:
I'm having a hard time figuring out what changed here. Let's revert the renames from this PR to retain the diffs, and then if you want to do the renames, add them as a separate PR.

Reply:
I'm not going to revert diffs just for code review purposes. We can sit at my desk and I can explain you what changed.

Reply:
The only func added was GetLargestStorageLocation and funcs it uses. The rest of the code is the same. Please review GetLargestStorageLocation.

Reply:
I don't want to set a precedent where code submitter has to rename files in order to facilitate easier code reviews. This will become messy very quickly.

The problem here is that git doesn't support rename but is using remove+add for renames. On top of that GitHub is using heuristics to determine if something is a rename or a new file. In this case, the changes are big enough that it decided it's a new file. 

I've created a separate commit with the changes to resource_location_retriever.go so you can review them easily:

https://github.com/GoogleCloudPlatform/compute-image-tools/pull/1044/commits/1ba9196e85275a66deb452bb020119b337d26719